soc/arm/microchip_mec/mec172x/reg/mec172x_regs.h

```diff
@@ -0,0 +1,1119 @@
+/*
+ * Copyright (c) 2009-2018 Arm Limited. All rights reserved.
+ * Copyright (c) 2021 Microchip Technology Inc. and its subsidiaries.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef MEC172X_REGS_H
+#define MEC172X_REGS_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * MEC172x includes the ARM single precicision FPU and the ARM MPU with
+ * eight regions.
+ */
+#define __MPU_PRESENT		1
+#define __FPU_PRESENT		1
```

Comment:
Can you add to your comment what these two are actually used for?

Also double underscore is normally reserved for compiler internals, as I understand it. Could you use MEC_ as a prefix instead, perhaps?

Reply:
Hi @sjg20 I moved these to soc.h they are required by ARM's CMSIS.  Zephyr headers do not define them.